### PR TITLE
Implement fulltext service for meta client

### PR DESCRIPTION
**Pull request by bright-starry-sky**
in progress. waiting storage side.

### CODE REVIEW

**Comment by bright-starry-sky** [comment]
Ready to review. thanks.

File: src/common/interface/meta.thrift [hunk 1]
```diff
@@ -826,6 +826,61 @@ struct ListIndexStatusResp {
...
+} (cpp.enum_strict)
+
+struct FTIndexItem {
+    1: required binary              index_name,
+    2: optional binary              index_template,
+    3: required FTIndexType         index_type,
+}
+
+struct AddFTHostsReq {
+    1: binary            name,
```
Comment (critical27) [inline src/common/interface/meta.thrift:847]: What is the usage of `name` field in. `AddFTHostsReq` and `RemoveFTHostsReq`? Is it an alias of hosts?

Comment (bright-starry-sky) [reply src/common/interface/meta.thrift:847]: > What is the usage of `name` field in. `AddFTHostsReq` and `RemoveFTHostsReq`? Is it an alias of hosts?

Used to register or unregister the full-text cluster, the name means cluster name. hosts means list of fulltext instance (ip:port) .

Comment (critical27) [reply src/common/interface/meta.thrift:847]: Ah, got it.

Comment (critical27) [reply src/common/interface/meta.thrift:847]: If so, do we need to specify `name` when create/drop full text index?

Comment (bright-starry-sky) [reply src/common/interface/meta.thrift:847]: > If so, do we need to specify `name` when create/drop full text index?

Good question, we need to specify `name` when create full text index. There may be two indexes in a space，tag index and edge index. We need to use this 'name' to identify them.

I also think the name 'AddFTHostsReq' is ambiguous,  is better using `signInFTService` and `signOutFTService` ?

**Comment by bright-starry-sky** [comment]
Addressed  @critical27's comments.

**Comment by bright-starry-sky** [comment]
Close this pr. unified code to #260

**Pull request closed without merging**